### PR TITLE
(de)serialization page draft.

### PR DESCRIPTION
**Pull request by gierlachg**
Draft of (de)serialization page.

### CODE REVIEW

**Comment by eminn** [comment]
I think it is better to limit chars in a single line to 80 in markdown files.

File: site/docs/api/serialization.md [hunk 1]
```diff
@@ -3,6 +3,87 @@ title: Serialization
...
 
-TODO
+To be able to send object state over a network or store it in a file 
+one has to first serialize it into raw bytes. Similarly, to be able to 
+fetch an object state over a wire or read it from a persistent storage 
+one has to deserialize it from raw bytes first. As Hazelcast Jet is a 
+distributed system by nature (de)serialization is integral part of it. 
+Understanding, when it is involved, how does it support the pipelines 
+and knowing differences between each of the strategies is crucial to 
+efficient Jet usage.
```
Comment (eminn) [inline site/docs/api/serialization.md]: WDYT  about `crucial to efficient usage of Hazelcast Jet` ?

File: site/docs/api/serialization.md [hunk 2]
```diff
@@ -3,6 +3,87 @@ title: Serialization
...
+
+Hazelcast Jet closely integrates with Hazelcast IMDG exposing many of 
+its features to Jet users. In particular, one can use IMDG data 
+structure as Jet `Source` and/or `Sink`. Objects retrieved from and 
+stored in those have to be (de)serializable.
+
+Another case which might require (de)serializable objects is sending 
+computation results between remote vertices. Hazelcast Jet tries to 
+minimize network traffic as much as possible, nonetheless different 
+parts of a `DAG` can reside on separate cluster members. To catch
```
Comment (eminn) [inline site/docs/api/serialization.md]: Maybe a link to the `DAG` section if we have it atm or a javadoc link ?

File: site/docs/api/serialization.md [hunk 3]
```diff
@@ -3,6 +3,87 @@ title: Serialization
...
+|     IdentifiedDataSerializable    | <ul><li>Relatively fast and space efficient</li></ul>                                                                                                | <ul><li>Requires implementation</li><li>Requires factory registration during cluster setup</li></ul> |
+|              Portable             | <ul><li>Faster and more space efficient than java standard interfaces</li><li>Supports versioning</li><li>Supports partial deserialization</li></ul> | <ul><li>Requires implementation</li><li>Requires factory registration during cluster setup</li></ul> |
+| [Stream&#124;ByteArray]Serializer | <ul><li>The fastest and lightest out of supported interfaces</li></ul>                                                                               | <ul><li>Requires implementation</li><li>Requires registration during cluster setup</li></ul>         |
+
+Below you can find rough performance numbers one can expect when 
+employing each of those strategies. A straightforward benchmark which continuously serializes and then
```
Comment (eminn) [inline site/docs/api/serialization.md]: Maybe a link to the benchmark source for reproducibility ?

Comment (gierlachg) [reply site/docs/api/serialization.md]: I'm not sure about this, I wanted the readers just to have an idea about the potential performance. Where would that sit if we wanted to add it?

Comment (eminn) [reply site/docs/api/serialization.md]: I think https://github.com/hazelcast/hazelcast-jmh is a good candidate for that. My thinking was they can take the benchmarking boilerplate and plug their own POJOs to see the real performance impact for their actual objects.

Comment (gierlachg) [reply site/docs/api/serialization.md]: Makes sense. I need to polish it then a bit and add it.

**Pull request merged**